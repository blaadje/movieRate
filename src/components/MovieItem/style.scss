.Item-wrapper {
  position: relative;
  width: 170px;
  height: 240px;
  -webkit-box-shadow: 0px 0px 0px 12px rgba(#2D3C4A, 0.8);
  -moz-box-shadow: 0px 0px 0px 12px rgba(#2D3C4A, 0.8);
  box-shadow: 0px 0px 0px 12px rgba(#2D3C4A, 0.8);
  margin-right: 2em;
  margin-bottom: 2em;
  cursor: pointer;
}

.Item-filter {
  position: relative;
  width: 100%;
  height: 100%;
  background: rgba(#176087, 0.2);
}

.Item-description {
  position: absolute;
  bottom: -3px;
  left: 0;
  width: 100%;
  padding: 1em;
  font-size: .5em;

  span {
    display: block;
  }
}

.Item-menu--wrapper {
  position: absolute;
  top: 0;
  bottom: -3px;
  width: 100%;
  background: rgba(#1E2E3D, 0.8);

  .Item-menu {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translateX(-50%) translateY(-50%);
  }

  .Item-menu--options {
    height: 50px;
    width: 50px;
    position: relative;
    background: rgba(#000000, 0.3);
    border-radius: 50%;
    margin-bottom: .5em;

    .Option-image {
      position: absolute;
      top: 50%;
      left: 50%;
      transform: translateX(-50%) translateY(-50%);
    }
  }
}

