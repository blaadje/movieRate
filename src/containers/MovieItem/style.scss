.Item-wrapper {
  position: relative;
  width: 160px;
  height: 240px;
  box-shadow: 0px 0px 0px 12px rgba(#2D3C4A, 0.8);
  margin: 1em;
  display: inline-block;
  margin-bottom: 3.2em;
  margin-top: 0;
  cursor: pointer;
}

.Item-filter {
  width: 100%;
  height: 100%;
  background: rgba(#176087, 0.2);
}

.Item-description {
  position: absolute;
  bottom: -3px;
  left: 0;
  width: 100%;
  padding: 1em;
  font-size: .5em;
  background: rgba(#000, 0.6);

  span {
    display: block;
  }
}

.Item-menu--wrapper {
  position: absolute;
  top: 0;
  bottom: -3px;
  width: 100%;
  background: rgba(#1E2E3D, 0.8);

  .Item-menu {
    z-index: 1;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translateX(-50%) translateY(-50%);
  }

  .Item-menu--options {
    height: 50px;
    width: 50px;
    position: relative;
    background: rgba(#000000, 0.3);
    border-radius: 50%;
    margin-bottom: .5em;

    .Option-image {
      position: absolute;
      top: 50%;
      left: 50%;
      transform: translateX(-50%) translateY(-50%);
    }
  }
}

