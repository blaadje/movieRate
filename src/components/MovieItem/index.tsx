import * as React from 'react'
import Svg from 'react-inlinesvg'

import * as checked from 'images/checked.svg'
import * as add from 'images/add.svg'

import BasePopper from 'components/Popper'

import './style.scss'

interface iProps {
  image: string,
  title: string,
  date: string,
  rate: number
}

interface iState {
  isHovered: Boolean
}

export default class MovieItem extends React.Component<iProps, iState> {
  constructor (props: iProps, state: iState) {
    super(props)
    this.state = {
      isHovered: false
    }
  }
  render () {
    const { isHovered } = this.state
    const background: Object = {
      backgroundRepeat: 'no-repeat',
      backgroundSize: 'cover'
    }

    return (
      <div
        className='Item-wrapper'
        style={{ background: `url(${this.props.image})`, ...background }}
        onMouseEnter={() => this.setState({ isHovered: true })}
        onMouseLeave={() => this.setState({ isHovered: false })}
      >
        <div className='Item-filter'>
          <div
            className='Item-description'
          >
            <span className='Item-description--title'>{this.props.title}</span>
            <span className='Item-description--date'>{`(${this.props.date})`}</span>
            <span className='Item-description--rate'>{this.props.rate}</span>
          </div>
        </div>
        {/* {isHovered && */}
          <div className='Item-menu--wrapper'>
            <div className='Item-menu'>
              <div className='Item-menu--options Option-rate'>
                <BasePopper
                  popperPlacement='right'
                  targetComponent={
                    <Svg className='Option-image' src={checked} />
                  }
                  popperComponent={
                    <div>test2</div>
                  }
                />
              </div>
              <div className='Item-menu--options Option-playlist'>
                <Svg className='Option-image' src={add} />
              </div>
            </div>
          </div>
        }
      </div >
    )
  }
}
