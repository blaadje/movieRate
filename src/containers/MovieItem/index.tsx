import * as React from 'react'

import ColumnItem from './modes/Column'
import RowItem from './modes/Row'
import './style/style.scss'

interface Iprops {
  isRow?: boolean
  movie: any
}

interface Istate {
  isHovered: Boolean
}

export default class MovieItem extends React.Component<Iprops, Istate> {
  constructor (props: Iprops, state: Istate) {
    super(props)
    this.state = {
      isHovered: false
    }
  }

  render (): any {
    const { isHovered } = this.state
    const { movie, isRow } = this.props

    //
    if (!isRow) {
      return <ColumnItem movie={movie} />
    }

    return (
      <RowItem
        onHovered={(value: boolean) => this.setState({ isHovered: value })}
        isHovered={isHovered}
        movie={movie}
      />
    )
  }
}
