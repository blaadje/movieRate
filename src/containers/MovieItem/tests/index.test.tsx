import MovieItem from '../'

import * as React from 'react'
const renderer = require('react-test-renderer')

it('renders correctly', () => {
  const tree = renderer
    .create(
      <MovieItem
        image='foo/bar'
        title='Pacific Rim'
        date='2018'
        rate={30}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
