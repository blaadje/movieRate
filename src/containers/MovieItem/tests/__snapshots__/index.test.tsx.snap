// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Image-wrapper Item-wrapper"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "background": "url(foo/bar)",
      "backgroundRepeat": "no-repeat",
      "backgroundSize": "cover",
    }
  }
>
  <div
    className="Item-filter"
  >
    <div
      className="Item-description"
    >
      <span
        className="Item-description--title"
      >
        Pacific Rim
      </span>
      <span
        className="Item-description--date"
      >
        (2018)
      </span>
      <span
        className="Item-description--rate"
      >
        30
      </span>
    </div>
  </div>
</div>
`;
